src/WorkItemMigrator/WorkItemImport/WitClient/WitClientUtils.cs

```diff
@@ -467,13 +467,27 @@ public void SaveWorkItemAttachments(WiRevision rev, WorkItem wi)
             // Save attachments
             foreach (WiAttachment attachment in rev.Attachments)
             {
-                if (attachment.Change == ReferenceChangeType.Added)
+                try
                 {
-                   AddSingleAttachmentToWorkItemAndSave(attachment, wi);
+                    if (attachment.Change == ReferenceChangeType.Added)
+                    {
+                       AddSingleAttachmentToWorkItemAndSave(attachment, wi);
+                    }
+                    else if (attachment.Change == ReferenceChangeType.Removed)
+                    {
+                        RemoveSingleAttachmentFromWorkItemAndSave(attachment, wi);
+                    }
                 }
-                else if (attachment.Change == ReferenceChangeType.Removed)
+                catch (Exception ex)
                 {
-                    RemoveSingleAttachmentFromWorkItemAndSave(attachment, wi);
+                    try
+                    {
+                        Logger.Log(ex, $"Failed to import attachment '{attachment.ToString()}'.");
+                    }
+                    catch (Exception)
+                    {
+                        throw ex; // yes, the ex from the nesting catch block
+                    }
```

Comment:
I'm not a huge fan of nested try-catch block. Why is this needed? What exception can be thrown from this line? `Logger.Log(ex, $"Failed to import attachment '{attachment.ToString()}'.")`
@bameije

Reply:
It's my way of safely adding stuff to code that I don't know: if I used the Logger.Log() in a wrong way (e.g. causing a null-ref exception or whatever), I want the program to behave exactly as before. Thus, not throwing the exception that I caused, but the original exception.

As you have requested this change in the main branch, we don't need this safety check anymore. I'll remove it.
```suggestion
                    Logger.Log(ex, $"Failed to import attachment '{attachment.ToString()}'.");
```